delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutesSpec.scala

```diff
@@ -346,6 +349,51 @@ class ResourcesRoutesSpec extends BaseRouteSpec with IOFromMap with CatsIOValues
       }
     }
 
+    "fail to update a schema without resources/write permission" in {
+      aclCheck.subtract(AclAddress.Root, Anonymous -> Set(resources.write)).accepted
+      Put(s"/v1/resources/$projectRef/_/someId/updateSchema") ~> routes ~> check {
+        response.status shouldEqual StatusCodes.Forbidden
+        response.asJson shouldEqual jsonContentOf("errors/authorization-failed.json")
+      }
+    }
+
+    "fail to update a schema when providing a non-existing schema" in {
+      aclCheck.append(AclAddress.Root, Anonymous -> Set(resources.write)).accepted
+      thereIsAResourceWithSchema { id =>
+        Put(s"/v1/resources/$projectRef/wrongSchema/$id/updateSchema") ~> routes ~> check {
+          response.status shouldEqual StatusCodes.NotFound
+          response.asJson.hcursor.get[String]("@type").toOption should contain("InvalidSchemaRejection")
+        }
+      }
+    }
+
+    "fail to update schema when providing _ as schema" in {
+      thereIsAResourceWithSchema { id =>
+        Put(s"/v1/resources/$projectRef/_/$id/updateSchema") ~> routes ~> check {
+          response.status shouldEqual StatusCodes.BadRequest
+          response.asJson.hcursor.get[String]("@type").toOption should contain("IdenticalSchema")
+        }
+      }
+    }
+
+    "fail to update schema when providing current schema" in {
```

Comment:
The remaining question is whether we want this behavior or not

Reply:
I don't think this provides any value to the user. 

A legitimate use case might be a user wanting to bulk update schemas, not knowing which ones might already be pointed at it already

If a user performed an update on a resource with an identical payload we wouldn't fail. We shouldn't fail here either. 

Not having this check is good for Delta because it's less code

Reply:
I'm resolving this thread because I have removed this test and the case where the schema is identical is handled by going to a fetch operation which is already testes.